src/Raven.Server/Documents/Indexes/CollectionOfBloomFilters.cs

```diff
@@ -95,14 +95,25 @@ internal unsafe BloomFilter CreateNewFilter(int number, Mode mode)
             Slice key;
             Slice.From(_context.Allocator, number.ToString("D9"), out key);
 
+            // we can safely pass raw pointers here and dispose DirectAdd scopes immediately because 
+            // filters' content will be written to overflows
+
             if (mode == Mode.X64)
             {
-                var ptr64 = _tree.DirectAdd(key, BloomFilter64.PtrSize);
-                return new BloomFilter64(key, ptr64, _tree, writeable: true);
+                Debug.Assert(_tree.ShouldGoToOverflowPage(BloomFilter64.PtrSize));
+
+                using (var ptr64 = _tree.DirectAdd(key, BloomFilter64.PtrSize))
+                {
+                    return new BloomFilter64(key, ptr64.Ptr, _tree, writeable: true);
+                }     
```

Comment:
Probably I am not understanding the solution, but I dont see any way that the addition of the scopes would solve the outstanding issue (Ptr may change along the ride). The new BloomFilter64 created here keeps the ptr alive outliving the scope, which by itself can become invalidated by an split the next sentence.